base/compiler/utilities.jl

```diff
@@ -138,8 +138,11 @@ use_const_api(li::CodeInstance) = invoke_api(li) == 2
 
 function get_staged(mi::MethodInstance, world::UInt)
     may_invoke_generator(mi) || return nothing
+    # enable caching of unoptimized generated code if the generator is `CachedGenerator`
+    cache_ci_ptr = (mi.def::Method).generator isa Core.CachedGenerator ?
+        pointer_from_objref(RefValue{CodeInstance}()) : C_NULL
```

Comment:
This is illegal. The `RefValue` is lost to the GC and the pointer is now considered leaked.

Reply:
What would be the best approach?
When this `cache_ci_ptr` is not null, a cache is generated and will be kept in the chain of `mi->cache`. Regardless, should this pointer be preserved at least during this `ccall`?